src/app/components/ExportModal.js

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import {
+    useContacts,
+    useUser,
+    useUserKeys,
+    useApi,
+    FormModal,
+    ResetButton,
+    PrimaryButton,
+    Alert
+} from 'react-components';
+import { queryContactExport } from 'proton-shared/lib/api/contacts';
+import downloadFile from 'proton-shared/lib/helpers/downloadFile';
+import { wait } from 'proton-shared/lib/helpers/promise';
+
+import { decryptContactCards } from '../helpers/decrypt';
+import { toICAL } from '../helpers/vcard';
+import { percentageProgress } from './../helpers/progress';
+import DynamicProgress from './DynamicProgress';
+
+const DOWNLOAD_FILENAME = 'protonContacts.vcf';
+// BACK-END DATA
+const QUERY_EXPORT_MAX_PAGESIZE = 50;
+const API_SAFE_INTERVAL = 100; // API request limit: 100 requests / 10 seconds, so 1 request every 100 ms is safe
+
+const ExportFooter = ({ loading }) => {
+    return (
+        <>
+            <ResetButton>{c('Action').t`Cancel`}</ResetButton>
+            <PrimaryButton loading={loading} type="submit">
+                {c('Action').t`Save`}
+            </PrimaryButton>
+        </>
+    );
+};
+
+const ExportModal = ({ onClose, ...rest }) => {
+    const api = useApi();
+    const [user] = useUser();
+    const [userKeysList, loadingUserKeys] = useUserKeys(user);
+    const [contacts, loadingContacts] = useContacts();
+
+    const [contactsExported, addSuccess] = useState([]);
+    const [contactsNotExported, addError] = useState([]);
+
+    const handleSave = (vcards) => {
+        const allVcards = vcards.join('\n');
+        const blob = new Blob([allVcards], { type: 'data:text/plain;charset=utf-8;' });
+        downloadFile(blob, DOWNLOAD_FILENAME);
```

Comment:
Add the current date with `moment` in the filename please.

Reply:
Which format do you prefer. I guess the standard would be:
```
const DOWNLOAD_FILENAME = `protonContacts-${moment().format()}.vcf`;
```
But I don't know if it will be too much gibberish for a normal user. Can't compare with v3 since that was simply producing `proton.vcf`